Add public `getTags` method

jennantilla opened the pull request:
# Description
## One Line Summary
Add public `OneSignal.User.getTags()` method to return local tags for the current user.

## Details
Add `getTags` public method that will return local tags for the current user on both Android and iOS.

### Motivation
We removed the `getTags` method in v5 of the SDKs to discourage use of OneSignal tags as a data store. We have heard feedback from app developers that they have a need for this method still, so we are adding back an implementation to return the local tags.

### Scope
Keep note it returns the local tags and not via a server fetch so any tags added mid-session through the REST API will not be reflected.

# Testing
## Unit testing
None

## Manual testing
Successfully call new method getTags on iPhone 15 emulator running iOS 17.0.1 and Pixel 7 running Android 14.

# Affected code checklist
   - [ ] Notifications
      - [ ] Display
      - [ ] Open
      - [ ] Push Processing
      - [ ] Confirm Deliveries
   - [ ] Outcomes
   - [ ] Sessions
   - [ ] In-App Messaging
   - [ ] REST API requests
   - [x] Public API changes

# Checklist
## Overview
   - [x] I have filled out all **REQUIRED** sections above
   - [x] PR does one thing
   - [x] Any Public API changes are explained in the PR details and conform to existing APIs

## Testing
   - [x] I have included test coverage for these changes, or explained why they are not needed
   - [x] All automated tests pass, or I explained why that is not possible
   - [x] I have personally tested this on my device, or explained why that is not possible

## Final pass
   - [x] Code is as readable as possible.
   - [x] I have reviewed this PR myself, ensuring it meets each checklist item

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/OneSignal/react-native-onesignal/1609)
<!-- Reviewable:end -->

jennantilla committed:
bump native sdk version dependencies to 5.0.5

jennantilla committed:
add getTags method

jennantilla committed:
add getTags button to example app

jennantilla committed:
Update migration guide to include getTags

nan-li commented on android/src/main/java/com/onesignal/rnonesignalandroid/RNOneSignal.java:
```diff
@@ -573,6 +575,20 @@ public void removeTags(ReadableArray tagKeys) {
+            promise.resolve(writableTags);
+        } catch (Throwable t) {
+            promise.reject(t.getMessage());
+        }
```
Looking at this again, there is no need for the try-catch. We would need it if we are working with JSON and need to handle possible JSONExceptions, but we aren't doing so here. 
I did notice that we have a lot of redundant try-catches in this file that can potentially hide problems. May be worth cleaning up in a future refactor.

jennantilla committed:
remove try/catch block from getTags

The try/catch block is unnecessary since creating a new JSONObject from a Map will not throw a JSONException.

rgomezp commented:
@emawby @nan-li @jennantilla why is getTags asynchronous here? This is not consistent with our other SDKs and with the description "returns local tags" https://documentation.onesignal.com/docs/mobile-sdk-reference#gettags

nan-li commented:
@rgomezp The call needs to go over the bridge to call the native function and receive the result. It's not consistent with the native SDKs since they don't need to do this.

rgomezp commented:
@nan-li I forgot about that limitation. Thanks!
